v3/src/models/data/data-set.ts

```diff
@@ -618,9 +618,44 @@ export const DataSet = V2Model.named("DataSet").props({
     Array.from(self.childCollection.caseGroupMap.values()).forEach(caseGroup => {
       self.itemIdChildCaseMap.set(caseGroup.childItemIds[0] ?? caseGroup.hiddenChildItemIds[0], caseGroup)
     })
+  },
+  // The returned array is in collection order.
+  // Includes cases with ANY child items selected, not just those with all items selected.
+  get selectedCaseIds(): Set<string>[] {
+    const selectedCaseIds: Set<string>[] = []
+    self.selection.forEach((itemId) => {
+      const caseIds = self.getItemCaseIds(itemId)
+      caseIds?.forEach((caseId, index) => {
+        const collection = self.collections[index]
+        if (!collection) return
+
+        if (!selectedCaseIds[index]) {
+          selectedCaseIds[index] = new Set<string>()
+        }
+        selectedCaseIds[index].add(caseId)
+      })
+    })
+
+    return selectedCaseIds
   }
 }))
 .views(self => ({
+  // The returned array includes arrays of the sorted indices of selected cases in each collection
+  // So selectedCaseIndices[i] contains an array of the indices of the cases selected in data.collections[i]
+  // Includes cases with ANY child items selected, not just those with all items selected.
+  get selectedCaseIndices(): number[][] {
```

Comment:
Following the logic of the previous comment, this would be `partiallySelectedCaseIndicesByCollection`.